### PR TITLE
Stackautomation

### PR DESCRIPTION
**Pull request by sshveta**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit acb9d06ad26f8f65b782d18f40be70b30b92fcd3:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 711a512770a32cbe4b8a1538a480305f2c782cc2:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4a146f59b80d316766b9487f4af7d61792627067:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 2686b416d113da9823c2770a3a3970f5278c3642:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/cloud/test_stack.py:6:1: F401 'settings' imported but unused
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0fa39227282931a48d8e52908372e5ae56cbebb2:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/cloud/test_stack.py:6:1: F401 'settings' imported but unused
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fead5ebcc5640300a5e52382a077944388211b29:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/cloud/test_stack.py:6:1: F401 'settings' imported but unused
```
:trollface:
*CFME QE Bot*

File: cfme/cloud/stack.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    pretty_attrs = ['name']
+
+    def __init__(self, name=None):
+        self.name = name
+
+    def load_details(self):
+        sel.force_navigate('clouds_stack', context={'stack': self})
+        flash.assert_no_errors()
+
+    def nav_to_security_group_link(self):
```
Comment (psav) [inline cfme/cloud/stack.py]: Shuldn't these be implemented as nav tree paths with context?

File: cfme/web_ui/__init__.py [hunk 2]
```diff
@@ -2020,6 +2020,7 @@ class Quadicon(Pretty):
             "no_snapshot": ("d", 'txt'),
             "policy": ("g", 'img'),
         },
+        "stack": {},
```
Comment (psav) [inline cfme/web_ui/__init__.py]: Do we have any quads at all?

File: cfme/tests/cloud/test_stack.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    sel.force_navigate("my_settings_default_views")
+    default_view = bg.active
+    if(default_view != 'Grid View'):
+        bg.choose('Grid View')
+        sel.click(form_buttons.save)
+
+
+def test_stack_detail_page(provider_init):
+    set_grid_view("Stacks")
+    stack = Stack(name="newstack")
```
Comment (psav) [inline cfme/tests/cloud/test_stack.py]: I'm assuming that ```stack``` will become a fixture at some point, probably module scoped?

File: cfme/web_ui/__init__.py (hunk 2 above)
Comment (sshveta) [inline cfme/web_ui/__init__.py]: No quads

File: cfme/tests/cloud/test_stack.py (hunk 3 above)
Comment (sshveta) [inline cfme/tests/cloud/test_stack.py]: Yes may be . Will make changes as i add more tests around stack and services.

Comment (psav) [inline cfme/tests/cloud/test_stack.py]: How does ```newstack``` get created?

Comment (sshveta) [inline cfme/tests/cloud/test_stack.py]: Right now , we can't create stack from CFME . I created this stack "newstack" in aws and hardcoded it for now

Comment (mfalesni) [inline cfme/tests/cloud/test_stack.py]: @sshveta So the name should be in YAML I think.

Comment (sshveta) [inline cfme/tests/cloud/test_stack.py]: Okay , will add in YAML.

**Comment by dajoRH** [comment]
Lint report for commit 0c0b85bec076b9745b71eeffca474b0738acc9bb:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/cloud/test_stack.py:6:1: F401 'settings' imported but unused
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f85465243e4c19cb31e6b91335a6702235cc0fe1:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/cloud/test_stack.py:6:36: E261 at least two spaces before inline comment
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0bfa7f0e4bf76b421879df5947ca9a33a09d70c4:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 23cd8609cbadb56b53de08bf935b51f3a8cbecfb:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**